### PR TITLE
Changelog: Output of template usage via `-vtemplates` has been improved

### PR DESCRIPTION
**Pull request by nordlow**

### CODE REVIEW

File: changelog/better-vtemplates.dd [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+declaration.
+
+All messages of this kind are sorted by descending unique instantiation count.
+
+If the flag argument `list-instances` is fed as
+
+    `-vtemplates=list-instances`
+
+the location for every instance of each instantiated template is printed along
+with information telling whether the instance was implicit or explicit.
```
Comment (Geod24) [inline changelog/better-vtemplates.dd:14]:
```suggestion
If the flag argument `list-instances` is passed (such as `-vtemplates=list-instances`),
each location where a template was instantiated is printed along with the reason for
the instantiation (implicit or explicit).
```

Formatting looked weird and a terminating newline was missing, but then I tried to rephrase this to be (IMO) a bit more readable. Take it or leave it, as long as the formatting is fixed.

Also, perhaps add an explanation / example of what an implicit instantiation as opposed to an explicit one.

Comment (nordlow) [reply changelog/better-vtemplates.dd]: Thanks. Done. Added a sample file aswell.

**Comment by nordlow** [comment]
macOS job needs to be restarted...

```
>>> TARGET FAILED: runnable/testptrref_gc.d
```

**Review by Geod24 — APPROVED** [review]

**Comment by wilzbach** [comment]
Releases are made from `stable` (this was master). Sorry, for not explicitly stating this on slack.

**Comment by nordlow** [comment]
So this will only go into next release? Do I have to do anything more?

**Comment by PetarKirov** [comment]
@nordlow we should cherry-pick this commit in stable like so:
```
git checkout stable
# Assuming 'upstream' points to this repo
git pull --ff-only upstream stable
git cherry-pick 603badf86a93ead1c4087ff860b6601132bce227

# Then open a PR on GitHub and target the stable branch
```

**Comment by Geod24** [comment]
Sorry, should've caught that by the fact it was a new file, not a modification 🤦

**Comment by nordlow** [comment]
> @nordlow we should cherry-pick this commit in stable like so:
> 
> ```
> # Make sure your local 'master' branch is up to date:
> git checkout master
> # (Assuming 'upstream' points to this repo)
> git pull --ff-only upstream master
> 
> # Make sure your local 'stable' branch is up to date:
> git checkout stable
> git pull --ff-only upstream stable
> 
> # Cherry pick the commit as it was merged into the 'master' branch:
> git cherry-pick 603badf86a93ead1c4087ff860b6601132bce227
> 
> # Then open a PR on GitHub and target the stable branch
> ```

Done: https://github.com/dlang/dmd/pull/11769/files
